# Finding the Best Hotel for the Perfect Charleston Vacation
### by Angel Fitzgerald, Capstone Project for the IBM Applied Data Science Certification

## Getting ZIP Codes and Associated Latitudes & Longitudes from Websites with BeautifulSoup
### ZIP Code Acquisition and Data Cleaning
To begin our exploration of the Charleston area, we will first use BeautifulSoup to scrape ZIP Codes from https://www.zip-codes.com/county/sc-charleston.asp.  This information will be transformed into a pandas dataframe.

In [1]:
#Import necessary libraries and webpage

import pandas as pd
import numpy as np

from bs4 import BeautifulSoup
import requests

page = requests.get('https://www.zip-codes.com/county/sc-charleston.asp#zipcodes').text
soup = BeautifulSoup(page, "lxml")

In [2]:
#Scrape table from web page

table = soup.find('table', class_='statTable')

In [3]:
#Get ZIP Code Data from table and store in list z

z = []

for zipcode in table.find_all('td', class_='label'):
    z.append(zipcode.text)

print(z)

['ZIP Code', 'ZIP Code 29401', 'ZIP Code 29402', 'ZIP Code 29403', 'ZIP Code 29404', 'ZIP Code 29405', 'ZIP Code 29406', 'ZIP Code 29407', 'ZIP Code 29409', 'ZIP Code 29412', 'ZIP Code 29413', 'ZIP Code 29414', 'ZIP Code 29415', 'ZIP Code 29416', 'ZIP Code 29417', 'ZIP Code 29418', 'ZIP Code 29419', 'ZIP Code 29422', 'ZIP Code 29423', 'ZIP Code 29424', 'ZIP Code 29425', 'ZIP Code 29426', 'ZIP Code 29429', 'ZIP Code 29439', 'ZIP Code 29449', 'ZIP Code 29451', 'ZIP Code 29455', 'ZIP Code 29457', 'ZIP Code 29458', 'ZIP Code 29464', 'ZIP Code 29465', 'ZIP Code 29466', 'ZIP Code 29470', 'ZIP Code 29482', 'ZIP Code 29487']


Now that the ZIP Code data is in a list, we can begin building and cleaning the pandas dataframe.

In [4]:
#Create a pandas dataframe from the ZIP Code list

df = pd.DataFrame(z) 
df.head()

0
0        ZIP Code
1  ZIP Code 29401
2  ZIP Code 29402
3  ZIP Code 29403
4  ZIP Code 29404

In [5]:
#The first row should be the label for the dataframe column.

new_header = df.iloc[0] #grab the first row for the label
df = df[1:] #take the data less the header row
df.columns = new_header #set the label row as the df label
df.head()

0        ZIP Code
1  ZIP Code 29401
2  ZIP Code 29402
3  ZIP Code 29403
4  ZIP Code 29404
5  ZIP Code 29405

In [6]:
#Finally, we should remove the text "ZIP Code" from each of the dataframe elements.  We do this by removing the first 8 characters of each ZIP Code element.

df['ZIP Code'] = df['ZIP Code'].map(lambda x: str(x)[8:])
df.head()

0 ZIP Code
1    29401
2    29402
3    29403
4    29404
5    29405

In [7]:
#According to the ZIP Codes website, we should have 34 ZIP Codes for Charleston.  Let's verify that by checking the shape of the dataframe.

df.shape

(34, 1)

It looks like we have the correct number of ZIP Codes for Charleston.

In [8]:
#Now, let's look at the data type for ZIP Code.

df.dtypes

0
ZIP Code    object
dtype: object

In [9]:
#Convert ZIP Code to int64.  Our ZIP Code data type will need to match for both the dataframe and the latitude & longitude data.

df = pd.to_numeric(df['ZIP Code'])
df.dtypes

dtype('int64')

### Latitude & Longitude .csv Acquisition and Appending this Data to the Dataframe
Next, latitudes and longitudes for the various neighborhoods are downloaded from https://public.opendatasoft.com/explore/dataset/us-zip-code-latitude-and-longitude/table/ as a .csv file.  These will be appended to the ZIP code dataframe.  

In [10]:
# The code was removed by Watson Studio for sharing.

In [11]:
#Read Latitude and Longitude .csv file.

Lat_Long = pd.read_csv(streaming_body_2, delimiter = ';')
Lat_Long.columns = ['ZIP Code', 'City', 'State', 'Latitude', 'Longitude', 'Time Zone', 'Daylight Savings Time Flag', 'Geopoint']
Lat_Long.head()

ZIP Code      City State   Latitude  Longitude  Time Zone  \
0     71937      Cove    AR  34.398483  -94.39398         -6   
1     72044  Edgemont    AR  35.624351  -92.16056         -6   
2     56171  Sherburn    MN  43.660847  -94.74357         -6   
3     49430    Lamont    MI  43.010337  -85.89754         -5   
4     52585  Richland    IA  41.194129  -91.98027         -6   

   Daylight Savings Time Flag             Geopoint  
0                           1  34.398483,-94.39398  
1                           1  35.624351,-92.16056  
2                           1  43.660847,-94.74357  
3                           1  43.010337,-85.89754  
4                           1  41.194129,-91.98027

In [12]:
#Remove City, State, Timezone, Daylight Savings Time Flag, and Geopoint from Lat_Long dataframe.

Lat_Long = Lat_Long.drop(columns=['City','State','Time Zone','Daylight Savings Time Flag', 'Geopoint'])
Lat_Long.head()

ZIP Code   Latitude  Longitude
0     71937  34.398483  -94.39398
1     72044  35.624351  -92.16056
2     56171  43.660847  -94.74357
3     49430  43.010337  -85.89754
4     52585  41.194129  -91.98027

In [13]:
#Before merging the latitude and longitude data with the Charleston ZIP Code dataframe, check the datatypes of latitude and longitude data.  
#Since we are merging ZIP Code data, this ZIP Code data must be of type int64, like the ZIP Code data in our dataframe.

Lat_Long.dtypes

ZIP Code       int64
Latitude     float64
Longitude    float64
dtype: object

In [14]:
#Now we can merge the Charleston ZIP Code dataframe with the latitude and longitude data.
result = pd.merge(df,
                 Lat_Long[['ZIP Code', 'Latitude', 'Longitude']],
                 on = 'ZIP Code')
result

ZIP Code   Latitude  Longitude
0      29401  32.779126  -79.93550
1      29402  32.848850  -79.85773
2      29403  32.799326  -79.94813
3      29404  32.897903  -80.06061
4      29405  32.856634  -79.98218
5      29406  32.918757  -80.02280
6      29407  32.794841  -80.00500
7      29409  32.848850  -79.85773
8      29412  32.737270  -79.95409
9      29413  32.848850  -79.85773
10     29414  32.821238  -80.05353
11     29415  32.848850  -79.85773
12     29416  32.848850  -79.85773
13     29417  32.848850  -79.85773
14     29418  32.879529  -80.05911
15     29419  32.848850  -79.85773
16     29422  32.848850  -79.85773
17     29423  32.981952  -80.07363
18     29424  32.783076  -79.93701
19     29425  32.786176  -79.94711
20     29426  32.790622  -80.38489
21     29429  32.972510  -79.66070
22     29439  32.662451  -79.93224
23     29449  32.715745  -80.26738
24     29451  32.798138  -79.76496
25     29455  32.717109  -80.08507
26     29457  32.848850  -79.85773
27     29458  33.105940  -79.46570
28     29464  32.813518  -79.85899
29     29465  32.848850  -79.85773
30     29466  32.876020  -79.79148
31     29470  32.788784  -80.22778
32     29482  32.763456  -79.83791
33     29487  32.651864  -80.17774

# Mapping the Neighborhoods in Charleston

In [15]:
#Import necessary libraries.

import json # library to handle JSON files

!conda install -c conda-forge geopy --yes
from geopy.geocoders import Nominatim # convert an address into latitude and longitude values

import requests # library to handle requests
from pandas.io.json import json_normalize # transform JSON file into a pandas dataframe

# Matplotlib and associated plotting modules
import matplotlib.cm as cm
import matplotlib.colors as colors

!conda install -c conda-forge folium=0.5.0 --yes 
import folium # map rendering library

print('Libraries imported.')

Solving environment: done

## Package Plan ##

  environment location: /opt/conda/envs/Python36

  added / updated specs: 
    - geopy


The following packages will be downloaded:

    package                    |            build
    ---------------------------|-----------------
    certifi-2019.6.16          |           py36_1         149 KB  conda-forge
    openssl-1.1.1c             |       h516909a_0         2.1 MB  conda-forge
    geographiclib-1.49         |             py_0          32 KB  conda-forge
    geopy-1.20.0               |             py_0          57 KB  conda-forge
    ca-certificates-2019.6.16  |       hecc5488_0         145 KB  conda-forge
    ------------------------------------------------------------
                                           Total:         2.5 MB

The following NEW packages will be INSTALLED:

    geographiclib:   1.49-py_0         conda-forge
    geopy:           1.20.0-py_0       conda-forge

The following packages will be UPDATED:

    ca-

In order to define an instance of the geocoder, define a user_agent. We will name our agent <em>charleston_explorer</em>, as shown below.

In [16]:
address = 'Charleston, SC'

geolocator = Nominatim(user_agent="charleston_explorer")
location = geolocator.geocode(address)
latitude = location.latitude
longitude = location.longitude
print('The geographical coordinates of Charleston are {}, {}.'.format(latitude, longitude))

The geographical coordinates of Charleston are 32.7876012, -79.9402728.


Create a map of Charleston with neighborhoods superimposed on top.

In [17]:
# create map of Charleston using latitude and longitude values
map_charleston = folium.Map(location=[latitude, longitude], zoom_start=11, control_scale = True)

# add markers to map
for lat, lng, z in zip(result['Latitude'], result['Longitude'], result['ZIP Code']):
    label = '{}'.format(z)
    label = folium.Popup(label, parse_html=True)
    folium.CircleMarker(
        [lat, lng],
        radius=5,
        popup=label,
        color='blue',
        fill=True,
        fill_color='#3186cc',
        fill_opacity=0.7,
        parse_html=False).add_to(map_charleston)  
    
map_charleston

## Exploring neighborhoods in Charleston.
### First, we will import Nearby Venues from Foursquare, based upon our location dataframe.

Define Foursquare Credentials and Version (Using Hidden Cell)

In [18]:
# The code was removed by Watson Studio for sharing.

Your credentials have been entered.


In [19]:
def getNearbyVenues(names, latitudes, longitudes, radius=500, LIMIT=100):
    
    venues_list=[]
    for name, lat, lng in zip(names, latitudes, longitudes):
        print(name)
            
        # create the API request URL
        url = 'https://api.foursquare.com/v2/venues/explore?&client_id={}&client_secret={}&v={}&ll={},{}&radius={}&limit={}'.format(
            CLIENT_ID, 
            CLIENT_SECRET, 
            VERSION, 
            lat, 
            lng, 
            radius, 
            LIMIT)
            
        # make the GET request
        results = requests.get(url).json()["response"]['groups'][0]['items']
        
        # return only relevant information for each nearby venue
        venues_list.append([(
            name, 
            lat, 
            lng, 
            v['venue']['name'], 
            v['venue']['location']['lat'], 
            v['venue']['location']['lng'],  
            v['venue']['categories'][0]['name']) for v in results])

    nearby_venues = pd.DataFrame([item for venue_list in venues_list for item in venue_list])
    nearby_venues.columns = ['Neighborhood', 
                  'Neighborhood Latitude', 
                  'Neighborhood Longitude', 
                  'Venue', 
                  'Venue Latitude', 
                  'Venue Longitude', 
                  'Venue Category']
    
    return(nearby_venues)

### Now write the code to run the above function on each neighborhood and create a new dataframe called *charleston_venues*.

In [20]:
charleston_venues = getNearbyVenues(names=result['ZIP Code'],
                                   latitudes=result['Latitude'],
                                   longitudes=result['Longitude']
                                    )

29401
29402
29403
29404
29405
29406
29407
29409
29412
29413
29414
29415
29416
29417
29418
29419
29422
29423
29424
29425
29426
29429
29439
29449
29451
29455
29457
29458
29464
29465
29466
29470
29482
29487


### How much data was retrieved for the Charleston area?  
#### We will check the total number of venues, and then we will return the total number of venues for each neighborhood.

In [21]:
#Checking size of resulting dataframe
print(charleston_venues.shape)
charleston_venues.head()

(307, 7)


Neighborhood  Neighborhood Latitude  Neighborhood Longitude  \
0         29401              32.779126                -79.9355   
1         29401              32.779126                -79.9355   
2         29401              32.779126                -79.9355   
3         29401              32.779126                -79.9355   
4         29401              32.779126                -79.9355   

                      Venue  Venue Latitude  Venue Longitude  \
0               Le Farfalle       32.780793       -79.934589   
1      Queen Street Grocery       32.777549       -79.936078   
2       Second State Coffee       32.779859       -79.937648   
3  Pure Barre Market Street       32.780714       -79.934621   
4               Fulton Five       32.779654       -79.933429   

       Venue Category  
0  Italian Restaurant  
1   Food & Drink Shop  
2         Coffee Shop  
3                 Gym  
4  Italian Restaurant

In [22]:
#Number of venues returned for each neighborhood.
charleston_venues.groupby('Neighborhood').count()

Neighborhood Latitude  Neighborhood Longitude  Venue  \
Neighborhood                                                         
29401                            65                      65     65   
29402                             4                       4      4   
29403                            15                      15     15   
29404                             4                       4      4   
29405                             7                       7      7   
29406                             3                       3      3   
29407                             4                       4      4   
29409                             4                       4      4   
29413                             4                       4      4   
29414                             4                       4      4   
29415                             4                       4      4   
29416                             4                       4      4   
29417                             4                       4      4   
29418                             2                       2      2   
29419                             4                       4      4   
29422                             4                       4      4   
29423                             1                       1      1   
29424                            81                      81     81   
29425                            27                      27     27   
29439                             4                       4      4   
29451                            12                      12     12   
29455                             8                       8      8   
29457                             4                       4      4   
29464                            15                      15     15   
29465                             4                       4      4   
29482                            15                      15     15   

              Venue Latitude  Venue Longitude  Venue Category  
Neighborhood                                                   
29401                     65               65              65  
29402                      4                4               4  
29403                     15               15              15  
29404                      4                4               4  
29405                      7                7               7  
29406                      3                3               3  
29407                      4                4               4  
29409                      4                4               4  
29413                      4                4               4  
29414                      4                4               4  
29415                      4                4               4  
29416                      4                4               4  
29417                      4                4               4  
29418                      2                2               2  
29419                      4                4               4  
29422                      4                4               4  
29423                      1                1               1  
29424                     81               81              81  
29425                     27               27              27  
29439                      4                4               4  
29451                     12               12              12  
29455                      8                8               8  
29457                      4                4               4  
29464                     15               15              15  
29465                      4                4               4  
29482                     15               15              15

## Results: The largest number of venues are in ZIP Code 29424, with 81 total venues.  The second largest number of venues are in ZIP Code 29401, with 65 total venues.  Interestingly, these ZIP Codes are right next to each other on the Charleston peninsula.  

### Let's reduce the dataframe so that we are only looking at the venues in 29424 and 29401.  We will call this dataframe *tourism_venues*.

In [24]:
tourism_venues_1 = charleston_venues.loc[charleston_venues.Neighborhood == 29424]
tourism_venues_2 = charleston_venues.loc[charleston_venues.Neighborhood == 29401]
tourism_venues = tourism_venues_1.append(tourism_venues_2) 
tourism_venues.tail()

Neighborhood  Neighborhood Latitude  Neighborhood Longitude  \
60         29401              32.779126                -79.9355   
61         29401              32.779126                -79.9355   
62         29401              32.779126                -79.9355   
63         29401              32.779126                -79.9355   
64         29401              32.779126                -79.9355   

                          Venue  Venue Latitude  Venue Longitude  \
60  Leyla Fine Lebanese Cuisine       32.783263       -79.934618   
61            Apple King Street       32.783382       -79.934854   
62             Chopsticks House       32.783331       -79.933750   
63              M. Dumas & Sons       32.783126       -79.934436   
64             FIRE street food       32.783097       -79.934457   

               Venue Category  
60  Middle Eastern Restaurant  
61          Electronics Store  
62         Chinese Restaurant  
63             Clothing Store  
64           Asian Restaurant

### Create a dataframe that only has hotels in 29424 and 29401. 

In [25]:
hotels = tourism_venues[tourism_venues['Venue Category'] == 'Hotel'].reset_index()
hotels

index  Neighborhood  Neighborhood Latitude  Neighborhood Longitude  \
0    159         29424              32.783076               -79.93701   
1    181         29424              32.783076               -79.93701   
2    195         29424              32.783076               -79.93701   
3    196         29424              32.783076               -79.93701   
4     11         29401              32.779126               -79.93550   
5     30         29401              32.779126               -79.93550   
6     33         29401              32.779126               -79.93550   
7     41         29401              32.779126               -79.93550   
8     55         29401              32.779126               -79.93550   

                                 Venue  Venue Latitude  Venue Longitude  \
0              The Restoration on King       32.782271       -79.934160   
1             Belmond Charleston Place       32.780998       -79.932466   
2                    Wentworth Mansion       32.779803       -79.939880   
3                     King Charles Inn       32.782630       -79.932240   
4             Belmond Charleston Place       32.780998       -79.932466   
5  The Mills House Wyndham Grand Hotel       32.777803       -79.931394   
6              The Restoration on King       32.782271       -79.934160   
7                    Wentworth Mansion       32.779803       -79.939880   
8                     King Charles Inn       32.782630       -79.932240   

  Venue Category  
0          Hotel  
1          Hotel  
2          Hotel  
3          Hotel  
4          Hotel  
5          Hotel  
6          Hotel  
7          Hotel  
8          Hotel

### Create a map of the hotels in 29424 and 29401.

In [29]:
# create map of Charleston using latitude and longitude values
map_29424 = folium.Map(location=[32.783076, -79.93701], zoom_start=18, control_scale = True)

#Create a list of latitude and longitude coordinate pairs.
locations = hotels[['Venue Latitude', 'Venue Longitude']]
locationlist = locations.values.tolist()
len(locationlist)

#add markers to map
for point in range(0, len(locationlist)):
    folium.Marker(locationlist[point], popup=hotels['Venue'][point]).add_to(map_charleston)
map_charleston

### Create a dataframe that has venues other than hotels in 29424 and 29401.

In [54]:
not_hotels = tourism_venues[tourism_venues['Venue Category'] != 'Hotel'].reset_index()
len(not_hotels)
not_hotels.head()

index  Neighborhood  Neighborhood Latitude  Neighborhood Longitude  \
0    137         29424              32.783076               -79.93701   
1    138         29424              32.783076               -79.93701   
2    139         29424              32.783076               -79.93701   
3    140         29424              32.783076               -79.93701   
4    141         29424              32.783076               -79.93701   

                               Venue  Venue Latitude  Venue Longitude  \
0                    Mellow Mushroom       32.783651       -79.934946   
1                   Caviar & Bananas       32.783780       -79.936034   
2                                 CO       32.784552       -79.935394   
3                      Basic Kitchen       32.782181       -79.934875   
4  Albert Simons Center for the Arts       32.784299       -79.936763   

          Venue Category  
0            Pizza Place  
1          Deli / Bodega  
2  Vietnamese Restaurant  
3             Restaurant  
4  College Arts Building

#### It makes sense that there are 137 venues that are not hotels.  We started with 146 and took away 9 hotels.  

### Map venues along with hotels to visualize venue density near hotels.

In [31]:
# create map of Charleston using latitude and longitude values
map_29424 = folium.Map(location=[32.783076, -79.93701], zoom_start=18, control_scale = True)

#Create a list of latitude and longitude coordinate pairs for hotels and not_hotels.
locations = hotels[['Venue Latitude', 'Venue Longitude']]
locationlist = locations.values.tolist()
not_locations = not_hotels[['Venue Latitude', 'Venue Longitude']]
not_locationlist = not_locations.values.tolist()

# add markers to map - hotels are markers, other venues are circles
for point in range(0, len(locationlist)):
    folium.Marker(locationlist[point], popup=hotels['Venue'][point]).add_to(map_charleston)
for point in range(0, len(not_locationlist)):
    folium.CircleMarker(not_locationlist[point], radius=5, color = 'RED').add_to(map_charleston)  

map_charleston

## Examining the map tells us that the three best hotels based upon walking distance to attractions, are The Restoration on King, Belmond Charleston Place, and King Charles Inn.  

## Let's determine the centroid of the venues to find out which hotel is closest to the most venues.

In [57]:
#Determine the centroid of the venue latitudes by finding their mean.

not_hotels.mean()

index                       113.255474
Neighborhood              29413.927007
Neighborhood Latitude        32.781346
Neighborhood Longitude      -79.936349
Venue Latitude               32.782308
Venue Longitude             -79.934154
dtype: float64

### The centroid's coordinates are (32.781346, -79.936349).  We can map this to see which hotel is closest to the centroid.

In [58]:
# create map of Charleston using latitude and longitude values
map_29424 = folium.Map(location=[32.783076, -79.93701], zoom_start=18, control_scale = True)

# Add centroid to map
folium.Marker(
    location=[32.781346, -79.936349], popup='Ideal Hotel Location', icon=folium.Icon(color='green')).add_to(map_charleston)

map_charleston

##  The best hotel, based upon number of venues within walking distance, is The Restoration on King.
### Let's examine the venue categories nearby.

In [59]:
# one hot encoding
tourism_onehot = pd.get_dummies(tourism_venues[['Venue Category']], prefix="", prefix_sep="")

# add neighborhood column back to dataframe
tourism_onehot['Neighborhood'] = tourism_venues['Neighborhood'] 

# move neighborhood column to the first column
fixed_columns = [tourism_onehot.columns[-1]] + list(tourism_onehot.columns[:-1])
tourism_onehot = tourism_onehot[fixed_columns]

tourism_onehot.head()

Neighborhood  American Restaurant  Antique Shop  Art Gallery  Art Museum  \
137         29424                    0             0            0           0   
138         29424                    0             0            0           0   
139         29424                    0             0            0           0   
140         29424                    0             0            0           0   
141         29424                    0             0            0           0   

     Arts & Crafts Store  Asian Restaurant  Athletics & Sports  Bakery  Bar  \
137                    0                 0                   0       0    0   
138                    0                 0                   0       0    0   
139                    0                 0                   0       0    0   
140                    0                 0                   0       0    0   
141                    0                 0                   0       0    0   

     ...  Shoe Store  Southern / Soul Food Restaurant  Sporting Goods Shop  \
137  ...           0                                0                    0   
138  ...           0                                0                    0   
139  ...           0                                0                    0   
140  ...           0                                0                    0   
141  ...           0                                0                    0   

     Sports Bar  Thai Restaurant  Theater  Vietnamese Restaurant  Wine Bar  \
137           0                0        0                      0         0   
138           0                0        0                      0         0   
139           0                0        0                      1         0   
140           0                0        0                      0         0   
141           0                0        0                      0         0   

     Wings Joint  Women's Store  
137            0              0  
138            0              0  
139            0              0  
140            0              0  
141            0              0  

[5 rows x 62 columns]

In [68]:
# Next, group rows by neighborhood and take the mean of the frequency of occurrence of each category
tourism_grouped = tourism_onehot.mean().reset_index()
tourism_grouped

index             0
0                      Neighborhood  29413.760274
1               American Restaurant      0.034247
2                      Antique Shop      0.006849
3                       Art Gallery      0.006849
4                        Art Museum      0.006849
5               Arts & Crafts Store      0.006849
6                  Asian Restaurant      0.013699
7                Athletics & Sports      0.006849
8                            Bakery      0.006849
9                               Bar      0.006849
10               Basketball Stadium      0.006849
11                  Bed & Breakfast      0.020548
12                        Bookstore      0.006849
13                         Boutique      0.041096
14                   Breakfast Spot      0.027397
15                     Burger Joint      0.013699
16                             Café      0.013699
17                      Candy Store      0.006849
18               Chinese Restaurant      0.013699
19                           Church      0.013699
20                   Clothing Store      0.047945
21                     Cocktail Bar      0.006849
22                      Coffee Shop      0.075342
23            College Arts Building      0.006849
24                   Cosmetics Shop      0.006849
25                    Deli / Bodega      0.006849
26                     Dessert Shop      0.013699
27                Electronics Store      0.013699
28                   Farmers Market      0.006849
29             Fast Food Restaurant      0.006849
..                              ...           ...
32                        Gift Shop      0.020548
33                     Gourmet Shop      0.013699
34                              Gym      0.027397
35                            Hotel      0.061644
36                        Hotel Bar      0.013699
37                   Ice Cream Shop      0.027397
38               Italian Restaurant      0.020548
39                   Lingerie Store      0.013699
40         Mediterranean Restaurant      0.013699
41               Mexican Restaurant      0.006849
42        Middle Eastern Restaurant      0.013699
43          New American Restaurant      0.013699
44                        Nightclub      0.013699
45                             Park      0.006849
46                         Pet Café      0.006849
47                      Pizza Place      0.013699
48                       Restaurant      0.034247
49                      Salad Place      0.006849
50                   Sandwich Place      0.020548
51               Seafood Restaurant      0.006849
52                       Shoe Store      0.013699
53  Southern / Soul Food Restaurant      0.054795
54              Sporting Goods Shop      0.027397
55                       Sports Bar      0.006849
56                  Thai Restaurant      0.006849
57                          Theater      0.006849
58            Vietnamese Restaurant      0.006849
59                         Wine Bar      0.006849
60                      Wings Joint      0.006849
61                    Women's Store      0.041096

[62 rows x 2 columns]

In [65]:
#Confirming new size
tourism_grouped.shape

(62, 2)

## There are 62 different types of venues within walking distance of the three recommended hotels.  Enjoy your Charleston vacation!